rust-toolchain.toml

```diff
@@ -0,0 +1,3 @@
+[toolchain]
+channel = "nightly-2022-08-08"
```

Comment:
What's the reason to fix such an old version? Can't we just remove this?

Reply:
@moCello we can absolutely remove in. In fact, I would prefer that.

I'm not sure why we version lock into a nightly, but I think the primary reason is due to its unstable nature. This CAN cause issues at times, but they're usually resolved in the next nightly anyway, or require a small change on our end.

I'll gladly turn it into `nightly` here and on JubJub.

Reply:
@moCello fixed it to `nightly` both here and in JubJub.

Clippy check has been enabled on both repositories, but the warnings do not block merging. The output of clippy check is visible [here](https://github.com/dusk-network/bls12_381/actions/runs/5119895221/jobs/9205725971#step:4:63). 

Depending on what you want to do with relation to clippy warnings and how the refactor goes, we can decide later on what to do with #77 & #104.